common/posix.edl

```diff
@@ -0,0 +1,395 @@
+            [out, count=d_namelen] char* d_name,
+            size_t d_namelen,
+            [out, count=1] int* err);
+
+        void oe_posix_rewinddir_ocall(
+            uint64_t dirp);
+
+        int oe_posix_closedir_ocall(
+            uint64_t dirp,
+            [out, count=1] int* err);
+
+        int oe_posix_stat_ocall(
+            [in, string] const char* pathname,
+            [out, count=1] struct oe_stat* buf,
+            [out, count=1] int* err);
+
+        int oe_posix_access_ocall(
+            [in, string] const char* pathname,
+            int mode,
+            [out, count=1] int* err);
+
+        int oe_posix_link_ocall(
+            [in, string] const char* oldpath,
+            [in, string] const char* newpath,
+            [out, count=1] int* err);
+
+        int oe_posix_unlink_ocall(
+            [in, string] const char* pathname,
+            [out, count=1] int* err);
+
+        int oe_posix_rename_ocall(
+            [in, string] const char* oldpath,
+            [in, string] const char* newpath,
+            [out, count=1] int* err);
+
+        int oe_posix_truncate_ocall(
+            [in, string] const char* path,
+            oe_off_t length,
+            [out, count=1] int* err);
+
+        int oe_posix_mkdir_ocall(
+            [in, string] const char* pathname,
+            oe_mode_t mode,
+            [out, count=1] int* err);
+
+        int oe_posix_rmdir_ocall(
+            [in, string] const char* pathname,
+            [out, count=1] int* err);
+
+       oe_host_fd_t oe_posix_socket_ocall(
+            int domain,
+            int type,
+            int protocol,
+            [out, count=1] int* err);
+
+        int oe_posix_socketpair_ocall(
+            int domain,
+            int type,
+            int protocol,
+            [in] oe_host_fd_t sv[2],
+            [out, count=1] int* err);
+
+        int oe_posix_connect_ocall(
+            oe_host_fd_t sockfd,
+            [in, size=addrlen] const struct oe_sockaddr* addr,
+            oe_socklen_t addrlen,
+            [out, count=1] int* err);
+
+        oe_host_fd_t oe_posix_accept_ocall(
+            oe_host_fd_t sockfd,
+            [in, out, size=addrlen_in] struct oe_sockaddr* addr,
+            oe_socklen_t addrlen_in,
+            [out, count=1] oe_socklen_t* addrlen_out,
+            [out, count=1] int* err);
+
+        int oe_posix_bind_ocall(
+            oe_host_fd_t sockfd,
+            [in, size=addrlen] const struct oe_sockaddr* addr,
+            oe_socklen_t addrlen,
+            [out, count=1] int* err);
+
+        int oe_posix_listen_ocall(
+            oe_host_fd_t sockfd,
+            int backlog,
+            [out, count=1] int* err);
+
+        ssize_t oe_posix_recvmsg_ocall(
+            oe_host_fd_t sockfd,
+            [out, size=msg_namelen] void* msg_name,
+            oe_socklen_t msg_namelen,
+            [out, count=1] oe_socklen_t* msg_namelen_out,
+            [out, size=msg_buflen] void* msg_buf,
```

Comment:
I just have a question here, is  struct iovec *msg_iov, int msg_iovlen etc serialized into msg_buf and deserialized later?

Reply:
Yes, that is correct. This is to work around the lack of a deep-copy feature in EDL. Instead of passing an array of iovs (not supported by EDL), we flatten the iov array into a single buffer and pass it to the host. The host constructs an iov array with a single element referring to msg_buf and msg_buflen.

Reply:
Deep-copy will come...!

Reply:
Yaay! :)